### PR TITLE
Bugfix - Due of incorrect bitmap processing devices with some `smartctl exit status` codes were missed

### PR DESCRIPTION
**Pull request by vagifzeynalov**

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -247,6 +247,7 @@ var (
 			"model_family",
 			"model_name",
 			"serial_number",
+			"message",
```
Comment (lahwaacz) [inline metrics.go:250]: Labeling the metric by an essentially arbitrary string is not the Prometheus way. See "Do not overuse labels" in the [documentation](https://prometheus.io/docs/practices/instrumentation/#things-to-watch-out-for). You're basically trying to put logs into Prometheus – but Prometheus is not a logging system.

Comment (vagifzeynalov) [reply metrics.go:250]: Sounds reasonable, thanks!
I'll remove it.

Comment (vagifzeynalov) [reply metrics.go]: @lahwaacz done

**Comment by baryluk** [comment]
> @lahwaacz FYI, I just found flags conversion from boolean
> 
> ```
>           "prefailure": true,
>           "updated_online": true,
>           "performance": true,
>           "error_rate": true,
>           "event_count": false,
>           "auto_keep": false
> ```
> 
> to string
> 
> ```
> flagsLong := smart.mineLongFlags(attribute.Get("flags"), []string{
> 			"prefailure",
> 			"updated_online",
> 			"performance",
> 			"error_rate",
> 			"event_count",
> 			"auto_keep",
> 		})
> ```
> 
> @baryluk ^^^ regarding "wasting storage, memory, CPU cycles and energy"

That does not really have any memory or storage impact in Prometheus when using default storage engine. The only thing that matter is number of time series with distinct labels, not actual values of the labels. Inclusion of these flags do not increase number of unique time series, just changes their labels, so there is essentially zero impact on storage.

File: readjson.go [hunk 2]
```diff
@@ -76,73 +73,23 @@ func readData(device string) (gjson.Result, error) {
...
-		if bits[7] == '1' {
-			logger.Warning("The device self-test log contains records of errors. [ATA only] Failed self-tests outdated by a newer successful extended self-test are ignored.")
-		}
-	}
-	return result
-}
-
 // Check json
-func jsonIsOk(json gjson.Result) bool {
+func jsonIsOk(json gjson.Result) {
```
Comment (lahwaacz) [inline readjson.go:86]: Is the function still actually used somewhere?

**Comment by SuperQ** [comment]
I think https://github.com/prometheus-community/smartctl_exporter/pull/37 was a simpler fix. Can you verify this?

**Pull request closed without merging**